### PR TITLE
Move and refactor integration-cli/registry to internal/test

### PR DESCRIPTION
**Pull request by vdemeester**
- Move the code from `integration-cli` to `internal/test`.
- Use `testingT` and `assert` when creating the registry.

:lion: 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: internal/test/registry/registry.go [hunk 1]
```diff
@@ -80,39 +94,51 @@ http:
...
 		password:    password,
 		email:       email,
-		registryURL: registryURL,
-	}, nil
+		registryURL: c.registryURL,
+	}
+}
+
+// WaitFor waits for the registry to be ready to serve requests (or fail after a while)
+func (r *V2) WaitFor(t testingT) {
```
Comment (cpuguy83) [inline internal/test/registry/registry.go]: Nit, `WaitReady` seems more appropriate here.

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by cpuguy83** [comment]
```
14:14:22 internal/test/registry/ops.go:22:6:warning: func name will be used as registry.RegistryURL by other packages, and that stutters; consider calling this URL (golint)
```

**Review by yongtang — APPROVED** [review]
LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Pull request merged**